AlpsSDK/ContextManager.swift

```diff
@@ -283,156 +261,114 @@ class ContextManager: NSObject, CLLocationManagerDelegate {
                         }
                         unknownBeacons.append(deviceId)
                     }
-                    break
                 }
             }
         }
     }
-    
+
     private func syncBeacon(beacon: CLBeacon) -> [IBeaconDevice] {
-        var b : [IBeaconDevice] = self.alpsManager.beacons
-        b = self.alpsManager.beacons.filter{
+        var b: [IBeaconDevice] = self.alpsManager.beacons
+        b = self.alpsManager.beacons.filter {
             let proximityUUID = $0.proximityUUID!
             let major = $0.major!
             let minor = $0.minor!
             // it will be called the number of time of beacons registered in the app. In example : It will be called 3 times because I have 3 beacons registered.
-            if (proximityUUID.caseInsensitiveCompare(beacon.proximityUUID.uuidString) == ComparisonResult.orderedSame)  && (major as NSNumber) == beacon.major && (minor as NSNumber) == beacon.minor {
+            if (proximityUUID.caseInsensitiveCompare(beacon.proximityUUID.uuidString) == ComparisonResult.orderedSame) && (major as NSNumber) == beacon.major && (minor as NSNumber) == beacon.minor {
                 return true
             }
             return false
         }
         return b
     }
-    
+
     private func triggerProximityEvent(userId: String, deviceId: String, proximityEvent: ProximityEvent, completion: @escaping (_ proximityEvent: ProximityEvent?) -> Void) {
         let userCompletion = completion
-        let _ = Alps.DeviceAPI.triggerProximityEvents(userId: userId, deviceId: deviceId, proximityEvent: proximityEvent) {
-            (proximityEvent, error) -> Void in
+        Alps.DeviceAPI.triggerProximityEvents(userId: userId, deviceId: deviceId, proximityEvent: proximityEvent) { (proximityEvent, _) in
             userCompletion(proximityEvent)
         }
     }
-    
+
     public func startBeaconsProximityEvent(forCLProximity: CLProximity) {
         proximityTrigger.insert(forCLProximity)
         // To change the TIMERS ! https://developer.apple.com/library/content/documentation/Performance/Conceptual/power_efficiency_guidelines_osx/Timers.html Reducing overhead
-//        switch forCLProximity {
```

Comment:
Yet, the timers for beacon proximity are too much energy consuming. We need to think of another way to remove undetected beacons.
I left it in comment, don't suppress it yet.

Reply:
Good catch. It was removed as a commented code.

I totally agree with energy consumption awareness 👍